sgx-types/src/ssa.rs

```diff
@@ -254,6 +316,37 @@ impl Footer {
             gpr,
         }
     }
+
+    /// Converts an internal fault type to a safe Rust enum describing the fault.
+    pub fn fault(&self) -> Option<Fault> {
+        let errcd = self.misc.exinfo.errcd;
+        let exitinfo = self.gpr.exitinfo;
+        let exit_type = exitinfo.exit_type();
+        let exception = exitinfo.exception();
+
+        if exit_type.is_none() || exception.is_none() {
+            return None;
+        }
+
+        match exitinfo.exception().unwrap() {
+            Exception::PageFault => {
+                let addr = self.misc.exinfo.maddr;
+                let pageflags = PageFault::from_bits(errcd).unwrap();
+                Some(Fault::PF {
+                    addr: addr,
+                    flags: pageflags,
+                })
+            }
+
+            Exception::GeneralProtection => {
+                let gpflags = GenProtFault::from_bits(errcd).unwrap();
+                Some(Fault::GP { flags: gpflags })
+            }
+            _ => {
+                return None;
+            }
+        }
+    }
```

Comment:
Try to reduce the number of temporary variables you are creating. Also, I think this code would benefit from this pattern:

```rust
match (self.gpr.exitinfo.exit_type(), self.gpr.exitinfo.exception()) {
    (Some(et), Some(Exception::PageFault)) => { ... }
    (Some(et), Some(Exception::GeneralProtection)) => { ... }
    _ => None,
}
```

Also,

Reply:
You're right, that is much more concise.